SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SchemaGenerator.cpp

```diff
@@ -484,33 +534,33 @@ void GenerateSubobjectSchemaForActor(FComponentIdGenerator& IdGenerator, UClass*
 
 	Writer.PrintNewLine();
 
-	GenerateActorIncludes(Writer, TypeInfo);
+	GenerateSubobjectSchemaForActorIncludes(Writer, TypeInfo);
 
 	FSubobjectMap Subobjects = GetAllSubobjects(TypeInfo);
 
 	bool bHasComponents = false;
 
 	for (auto& It : Subobjects)
 	{
-		uint32 Offset = It.Key;
 		TSharedPtr<FUnrealType>& SubobjectTypeInfo = It.Value;
 		UClass* SubobjectClass = Cast<UClass>(SubobjectTypeInfo->Type);
 
-		FSubobjectSchemaData SubobjectData;
+		FActorSpecificSubobjectSchemaData SubobjectData;
 
-		if (IsReplicatedSubobject(SubobjectTypeInfo) && SchemaGeneratedClasses.Contains(SubobjectClass))
+		if (SchemaGeneratedClasses.Contains(SubobjectClass))
 		{
 			bHasComponents = true;
-			SubobjectData = GenerateSubobjectSpecificSchema(Writer, IdGenerator, UnrealNameToSchemaComponentName(SubobjectTypeInfo->Name.ToString()), SubobjectTypeInfo, SubobjectClass, ActorClass, Offset);
+			SubobjectData = GenerateSubobjectSpecificSchema(Writer, IdGenerator, UnrealNameToSchemaComponentName(SubobjectTypeInfo->Name.ToString()), SubobjectTypeInfo, SubobjectClass, ActorClass, 0);
 		}
 		else
 		{
 			SubobjectData.ClassPath = SubobjectClass->GetPathName();
 		}
 
 		SubobjectData.Name = SubobjectTypeInfo->Name;
-		ActorSchemaData.SubobjectData.Add(Offset, SubobjectData);
-		ClassPathToSchemaData.Add(SubobjectClass->GetPathName(), FSchemaData());
+		uint32 SubobjectOffset = SubobjectData.SchemaComponents[SCHEMA_Data];
+		check(SubobjectOffset != 0);
```

Comment:
Do we want to crash the editor if this happens? Probably error out no?

Reply:
Will leave this here because the check is only ever hit when we write incorrect code. If code is correct, the check won't be hit.